### PR TITLE
Vs2015 fixes

### PR DESCRIPTION
**Pull request by juj**
Fixes a build error (std::isdigit -> isdigit) on Windows, and the rest are about removing build warnings.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,8 +29,11 @@ SET(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/lib")
 SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/lib")
 
 IF(MSVC)
-  ADD_COMPILE_FLAG("/arch:sse2")
-  ADD_COMPILE_FLAG("/Wall")
+  IF(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "19.0") # VS2013 and older explicitly need /arch:sse2 set, VS2015 no longer has that option, but always enabled. 
+    ADD_COMPILE_FLAG("/arch:sse2")
+  ENDIF()
+  ADD_COMPILE_FLAG("-D_CRT_SECURE_NO_WARNINGS")
```
Comment (jfbastien) [inline CMakeLists.txt]: Could you open a bug to fix this, and remove this define? I don't think we should be using these functions :-)

File: src/support/file.cpp [hunk 2]
```diff
@@ -27,8 +27,13 @@ T wasm::read_file(const std::string &filename, bool debug) {
     exit(EXIT_FAILURE);
   }
   infile.seekg(0, std::ios::end);
-  size_t insize = infile.tellg();
-  T input(insize + 1, '\0');
+  std::streamoff insize = infile.tellg();
+  if (sizeof(size_t) == 4 && insize >= 0xFFFFFFFFU) {
```
Comment (jfbastien) [inline src/support/file.cpp]: Use numeric_limits here instead of hard-coding these values.

File: src/support/file.cpp [hunk 3]
```diff
@@ -27,8 +27,13 @@ T wasm::read_file(const std::string &filename, bool debug) {
     exit(EXIT_FAILURE);
   }
   infile.seekg(0, std::ios::end);
-  size_t insize = infile.tellg();
-  T input(insize + 1, '\0');
+  std::streamoff insize = infile.tellg();
```
Comment (jfbastien) [inline src/support/file.cpp]: This should be `std::streampos`, but it's probably better to use `auto` here.

File: src/wasm-s-parser.h [hunk 4]
```diff
@@ -23,6 +23,7 @@
 #define wasm_wasm_s_parser_h
 
 #include <cmath>
+#include <ctype.h>
```
Comment (jfbastien) [inline src/wasm-s-parser.h]: `cctype` gives you `std::isdigit`.

**Comment by kripken** [comment]
Very nice work, thanks!

lgtm, no additional comments on top of @jfbastien's.

File: CMakeLists.txt (hunk 1 above)
Comment (juj) [inline CMakeLists.txt]: Sure, #283.

File: src/support/file.cpp (hunk 2 above)
Comment (juj) [inline src/support/file.cpp]: Followed a code pattern in existing location in code, changed that to use numeric_limits as well.

File: src/support/file.cpp (hunk 3 above)
Comment (juj) [inline src/support/file.cpp]: Changed to `std:streampos`

File: src/wasm-s-parser.h (hunk 4 above)
Comment (juj) [inline src/wasm-s-parser.h]: Check.

**Comment by juj** [comment]
Thanks, addressed review comments.

**Comment by kripken** [comment]
Looks like that streamoff stuff breaks on travis, https://travis-ci.org/WebAssembly/binaryen/jobs/119046057

**Comment by jfbastien** [comment]
May as well compare to `ssize_t`: it's a lot of memory anyways :-)

**Comment by kripken** [comment]
Rebased, fixed last issue, and merged.

**Comment by juj** [comment]
Thanks for merging and fixing the error!

**Pull request merged**